spokestack/wakeword/tflite.py

```diff
@@ -94,87 +103,46 @@ def __call__(self, context: SpeechContext, frame) -> None:
         Returns: None
 
         """
+        vad_fall = self._is_speech and not context.is_speech
+        self._is_speech = context.is_speech
         if not context.is_active:
-            self.sample(context, frame)
+            self._sample(context, frame)
 
-        if not context.is_speech:
+        if vad_fall:
             self.reset()
 
-    def sample(self, context: SpeechContext, frame) -> None:
-        """ Iterates over samples from an audio frame
-
-        Args:
-            context (SpeechContext): current state of the speech pipeline
-            frame (np.ndarray): a single frame of an audio signal
+    def _sample(self, context: SpeechContext, frame) -> None:
 
-        Returns: None
+        frame /= 2 ** 15 - 1
+        frame = np.clip(frame, -1.0, 1.0)
+        frame = np.append(frame[0], frame[1:] - self.pre_emphasis * frame[:-1])
 
-        """
         for sample in frame:
-            # normalize
-            sample = sample / frame.max()
-            # clip
-            sample = np.clip(sample, -1.0, 1.0)
-            # preemphasis
-            prev_sample = frame[-1]
-            sample -= self.pre_emphasis * np.hstack([[self.prev_sample], sample])[:-1]
-            self.prev_sample = prev_sample
 
             self.sample_window.write(sample)
+
             if self.sample_window.is_full:
                 if context.is_speech:
                     self._analyze(context)
                 self.sample_window.rewind().seek(self.hop_length)
 
     def _analyze(self, context: SpeechContext) -> None:
-        """ Applies preprocessing to a frame of audio
-
-        Args:
-            context (SpeechContext): current state of the speech pipeline
-
-        Returns: None
-
-        """
-
-        frame = self.sample_window.to_array()
-        window = signal.get_window("hann", self.window_size)
-        frame = np.fft.rfft(frame * window, n=self.window_size)
+        frame = self.sample_window.read_all()
+        frame = np.fft.rfft(frame * self._hann_window, n=self.window_size)
         frame = np.abs(frame)
-        self.sample_window.reset()
 
-        self.fft_window.write(frame)
+        self.fft_frame[0] = frame
```

Comment:
no need for a member/assignment here, the `frame` variable can just be passed to the `_filter` function